src/block_disagg/block_disagg_write.c

```diff
@@ -111,10 +110,14 @@ __wt_block_disagg_write_internal(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *bloc
 
     /* Get the page ID. */
     page_id = block_meta->page_id;
-
     /* Get the checkpoint ID. */
     WT_ACQUIRE_READ(checkpoint_id, conn->disaggregated_storage.global_checkpoint_id);
-
+    /*
+     * TODO: this assertion holds now but will need to change when we fix checkpoint id assignment
```

Comment:
In which case it would be okay for these checkpoint IDs to not match?

The current architecture does not allow us to write pages with different checkpoint IDs. At the beginning of a checkpoint, PALI calls `pl_begin_checkpoint` with a new checkpoint ID, then we write pages with that checkpoint ID, and then we call `pl_complete_checkpoint` with the same ID.

For example, if we are doing a global checkpoint with ID 10, and after we finish checkpointing table A as a part of the global checkpoint, we cannot evict pages from it with the next checkpoint ID 11, because that would come _before_ we begin checkpoint 11. SLS-486 describes this issue and calls for us to explicitly prohibit this behavior.

If we'd like to be able to write pages from two different checkpoints, we would need to bring it up with the DEs and the rest of the team to see what the implications would be.

Reply:
Thanks for the explanation.